.github/workflows/build_library.yml

```diff
@@ -47,12 +47,12 @@ jobs:
       matrix:
         include:
           - target: x86_64-apple-darwin
-            py-platform: macosx-10_15_x86_64
+            py-platform: macosx-14_0_x86_64
```

Comment:
Do we require >= macOS 14 now for all of sentry development? If not, I'd at least keep the intel build where it was. 

Technically, there shouldn't be any problems, I noticed that even built on newer versions binaries require a lower version number of the system libraries, but still...

Reply:
the value here ~generally needs to match the platform it is built from -- it's already wrong on master :)

but yes we already require macos>=14